src/main/java/seedu/address/logic/parser/ViewCommandParser.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+
+import seedu.address.logic.commands.ViewCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.Date;
+import seedu.address.model.appointment.DateMatchesPredicate;
+
+/**
+ * Parse input arguments and creates a new ViewCommand object.
+ */
+public class ViewCommandParser implements Parser<ViewCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ViewCommand
+     * and returns a ViewCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    @Override
+    public ViewCommand parse(String arg) throws ParseException {
+        requireNonNull(arg);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(arg, PREFIX_DATE);
+
+        if (argMultimap.getValue(PREFIX_DATE).isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_USAGE)
+            );
```

Comment:
Shall we consider the situation when a nurse wants to view the whole appointment list (without a date specified)?

Reply:
I think that he handled this under `NuudleParser` but I think it would be better to do the argument checking in this file instead?
Maybe can try doing the `isEmpty` check before `tokenize`?

Reply:
Hmm, as @YangYue128-helen mentioned, perhaps it'll be ok to leave out this check as we are able to support view commands without a `DATE ` predicate as well? 

Based on line 53 of `ViewCommand.java`, the current system seems to support the display of all future appointments if the predicate input is null (in the case that a date is not entered by the user) so I think you've got it covered @ZhangWanlin98 👍🏻

Reply:
Oh wait after reviewing `NuudleParser` as mentioned by @JinHao-L , I think the check done here is actually valid as the ViewCommandParser is only instantiated if extra input by the user following the `view` command is detected. Hence, it actually indicates an invalid command format if `PREFIX_DATE `is empty as it would mean that something else is written.

Reply:
Hence, it actually indicates an invalid command format if PREFIX_DATE is empty as it would mean that something else is written
---------------------------------------------------------------------------------------------------------------------------------------
@Avalionnet yup, I covered empty arg in the switch cases in NuudleParser, which is just list other one word command
but now I have changed to cover it in the ViewCommandParser
I think both ways are good, what yall think?